prov/efa/src/rdm/efa_rdm_pke_rtm.c

```diff
@@ -137,7 +137,14 @@ ssize_t efa_rdm_pke_init_rtm_with_payload(struct efa_rdm_pke *pkt_entry,
 	if (txe->internal_flags & EFA_RDM_OPE_READ_NACK)
 		rtm_hdr->flags |= EFA_RDM_REQ_READ_NACK;
 
-	if (data_size == -1) {
+	/* If this RTM packet is sent after the runting read protocol has failed
+	because of a MR registration limit on the receiver, we don't want to
+	send any data with the RTM packet. This is because the runting read RTM
+	packets have already delivered some of the data and the long CTS RTM
+	packet does not have a seg_offset field */
+	if (txe->internal_flags & EFA_RDM_OPE_READ_NACK) {
```

Comment:
so it is because long cts rtm currently doesn't support sending from a non-zero offset. data pkts supports offset so it is not impacted. Do I understand correctly?

Reply:
Yes, exactly